README.md

```diff
@@ -26,13 +26,50 @@ And don't forget to check our [Awesome Embedded Rust][aer] list! The thing you a
 
 * Useful Links
     * [Contributing Guide]
-    * [Item Template](https://github.com/rust-embedded/not-yet-awesome-embedded-rust#not-yet-awesome-item-template)
+    * [Item Template](#not-yet-awesome-item-template)
 * Not Yet Awesome List
-    * [nothing yet...](#)
+    * [Sharing Data With Interrupts](#sharing-data-with-interrupts)
 
 # The List
 
-Nothing here yet...
+## Sharing Data With Interrupts
+
+### Background
+
+Currently, it is not convenient to share data with an interrupt using safe Rust. Because interrupt handlers are functions that take no arguments, all data consumed by interrupts must either be local `static` variables, or global `static` variables.
+
+Global variables are not great in Rust, because:
+
+* All mutable access must be `unsafe`
+* Not all data can be initialized in a `const` context, so it's often necessary to use an `Option<T>`
+* Global variables aren't typically idiomatic Rust.
+
+Tools like [cortex-m-rtfm] achieve this in a zero cost fashion by using a Domain Specific Language to automatically provide safe access to shared resources between tasks and interrupts, however these tools can not be used by applications not using RTFM, or by libraries such as HAL or BSP crates.
```

Comment:
Maybe also mention atomics? Or it could go in the patterns book. Atomics in `static`s are fine if you scope them; you can use them in libraries as well. For example

``` rust
// crate: my-hal-impl

mod timer {
    // not global
    static DONE: AtomicBool = AtomicBool::new(false);

    impl Timer {
        fn wait(&mut self, cycles: u32) {
            DONE.store(false, Ordering::Relaxed);

            // set a timeout

            while !DONE.load(Ordering::Relaxed) {
                asm::wfi();
            }
        }
    }

    #[interrupt]
    fn TIM2() {
        // clear interrupt flag

        DONE.store(true, Ordering::Relaxed);
    }
}
```

Reply:
(Also I would change "tools like ..." to "frameworks like ...")

Reply:
> Maybe also mention atomics?

Or you could reword the request to say "share non-atomic data ..."

Reply:
> (Also I would change "tools like ..." to "frameworks like ..." (in the text))

I actually went back and forth on that exact naming difference! Thanks for the deciding factor.

> Maybe also mention atomics? Or it could go in the patterns book. Atomics in `static`s are fine if you scope them; you can use them in libraries as well.

I'll add a bit about this, once there is a section about atomics in the patterns book, I'd be happy to link it here as well.

I think we will need to make a folder of "detailed requests" sooner than later - I'd like to keep the README from becoming 10k-100k lines long, so at some point we should probably move each NYA item to it's own MD file, and just have a title and 1 paragraph summary (and a link to the child MD) on the main readme.